imap-proto/src/types.rs

```diff
@@ -71,6 +71,10 @@ pub enum ResponseCode<'a> {
     AppendUid(u32, Vec<UidSetMember>),
     CopyUid(u32, Vec<UidSetMember>, Vec<UidSetMember>),
     UidNotSticky,
+    MetadataLongEntries(u64), // RFC 5464, section 4.2.1
+    MetadataMaxSize(u64),     // RFC 5464, section 4.3
+    MetadataTooMany,          // RFC 5464, section 4.3
+    MetadataNoPrivate,        // RFC 5464, section 4.3
```

Comment:
Any thoughts on whether this ought to be a single variant that contains another enum value? Perhaps that would make it easier to handle these as a group if desired?

Reply:
I don't _think_ they should be split out. They're not generally matched together, and if they were, it'd be _either_ the first _or_ the last three. And it seems awkward to have an enum that covers just the last three, or one that covers all four when only three of them will be matched as a group..